### PR TITLE
Us014 create a new band and add it to job role

### PR DESCRIPTION
**Pull request by BartlomiejMilewski**
### 🛠 Changes being made

Here give examples of the changes you've made in this pull request. Include an itemized list if you can. It'll help the reviewer

### ✨ Definition of Done (all the points need to be filled out before merging)

- [ ] Meets Acceptance Criteria
- [ ] Tested
- [ ] Code Review made by team 2 members
- [ ] UX approved
- [ ] Product Owner Approved

### 🧪 Non-Functional Requirements

- [ ] Automated tests should cover acceptance criterias (Unit tests, Integration/API/Component tests)
- [ ] All input must be validated (client side & API)
- [ ] Sensitive data should not be stored in the URL
- [ ] No secrets in codesource

### CODE REVIEW

File: product-backend/scripts/populateDataBase.sh [hunk 1]
```diff
@@ -1 +1,2 @@
-mysql -h"$DB_HOST" -D"$DB_NAME" -u"$DB_USERNAME" -p"$DB_PASSWORD" < "populate-database-job-roles.sql"
\ No newline at end of file
+mysql -h"$DB_HOST" -D"$DB_NAME" -u"$DB_USERNAME" -p"$DB_PASSWORD" < "populate-database-job-roles.sql"
+mysql -h"$DB_HOST" -D"$DB_NAME" -u"$DB_USERNAME" -p"$DB_PASSWORD" < "populate-database-job-bands.sql"
```
Comment (ljercha) [inline product-backend/scripts/populateDataBase.sh]: Job roles depends on bands
```suggestion
mysql -h"$DB_HOST" -D"$DB_NAME" -u"$DB_USERNAME" -p"$DB_PASSWORD" < "populate-database-job-bands.sql"
mysql -h"$DB_HOST" -D"$DB_NAME" -u"$DB_USERNAME" -p"$DB_PASSWORD" < "populate-database-job-roles.sql"
```

File: product-backend/src/main/java/org/kainos/ea/controller/BandController.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.SQLException;
+
+@Tag(name = "Job Band")
+@Path("/api")
+
+public class BandController {
+    private final static Logger logger = LoggerFactory.getLogger(BandService.class);
```
Comment (ljercha) [inline product-backend/src/main/java/org/kainos/ea/controller/BandController.java]:
```suggestion
    private final static Logger logger = LoggerFactory.getLogger(BandController.class);
```

File: product-backend/src/main/java/org/kainos/ea/controller/BandController.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            String errorMessage = "Band already exists!";
+            ErrorResponse errorResponse = new ErrorResponse(errorMessage);
+            return Response.status(Response.Status.BAD_REQUEST).entity(errorResponse).build();
+        }
+    }
+
+    @GET
+    @Path("/band/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getBandById(@PathParam("id") int id) {
```
Comment (ljercha) [inline product-backend/src/main/java/org/kainos/ea/controller/BandController.java]: do you need this API (including DAO method getBandById)?

File: product-web-app/tests/bandService.test.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    });
+
+after(() => {
+      logger.unsilent();
+    });
+
+
+    describe('createNewBand', () => {
+        it('When API online expect Band to be created', async () => {
+            const mockAxios = new MockAdapter(axios);
```
Comment (ljercha) [inline product-web-app/tests/bandService.test.ts]: import mockAxios from https://github.com/Fluffyguy611/agile-project-team-A/blob/main/product-web-app/tests/axios.instance.test.ts

File: product-web-app/tests/bandService.test.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            const responseBody = await bandService.createNewBand(testBandEngi);
+
+            expect(responseBody).to.deep.equal(testBandEngi);
+            sinon.assert.calledOnceWithExactly(bandValidatorStub.validateBand, testBandEngi);
+
+            mockAxios.restore();
+        });
+
+        it('When Band has invalid fields expect exception', async () => {
+            const mockAxios = new MockAdapter(axios);
```
Comment (ljercha) [inline product-web-app/tests/bandService.test.ts]: same as above

File: product-web-app/tests/jobRoleService.test.ts [hunk 6]
```diff
@@ -7,21 +7,22 @@ import JobRole from '../model/jobRole.js';
 import JobRoleService from '../service/jobRoleService.js';
 import logger from '../service/logger.js';
 
-const mockAxios = new MockAdapter(axios);
```
Comment (ljercha) [inline product-web-app/tests/jobRoleService.test.ts]: import from common file
https://github.com/Fluffyguy611/agile-project-team-A/blob/main/product-web-app/tests/axios.instance.test.ts

File: product-web-app/views/add-new-job-role-band.html [hunk 7]
```diff
@@ -0,0 +1,26 @@
+{% extends "./parent.html" %}
+{% block content %}
```
Comment (ljercha) [inline product-web-app/views/add-new-job-role-band.html]: rename the file

File: product-web-app/views/view-single-band.html [hunk 8]
```diff
@@ -0,0 +1,49 @@
+{% extends "./parent.html" %}
```
Comment (ljercha) [inline product-web-app/views/view-single-band.html]: There is no requirement to see a single band

**Review by ljercha — CHANGES_REQUESTED** [review]

File: product-backend/src/main/java/org/kainos/ea/controller/BandController.java (hunk 3 above)
Comment (BartlomiejMilewski) [reply product-backend/src/main/java/org/kainos/ea/controller/BandController.java]: Removed method and endpoint

File: product-backend/src/test/java/tests/BandValidatorTests.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.kainos.ea.model.BandRequest;
+import org.kainos.ea.service.BandValidator;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(MockitoExtension.class)
+public class BandValidatorTests {
```
Comment (ljercha) [inline product-backend/src/test/java/tests/BandValidatorTests.java:14]: You are missing a few unit test cases

File: product-backend/src/main/java/org/kainos/ea/exception/BandDoesNotExistException.java [hunk 10]
```diff
@@ -0,0 +1,4 @@
+package org.kainos.ea.exception;
+
+public class BandDoesNotExistException extends Exception {
```
Comment (ljercha) [inline product-backend/src/main/java/org/kainos/ea/exception/BandDoesNotExistException.java]: not used

**Review by ljercha — COMMENTED** [review]
Fix those and I am happy to approve

**Review by ljercha — APPROVED** [review]

File: product-backend/src/main/java/org/kainos/ea/exception/BandAlreadyExistsException.java
Comment (mtslech) [inline product-backend/src/main/java/org/kainos/ea/exception/BandAlreadyExistsException.java:1]: @Override
    public String getMessage() {
        return "Band already exists";
    }

could override getMessage method

File: product-backend/src/main/java/org/kainos/ea/service/BandValidator.java [hunk 11]
```diff
@@ -0,0 +1,26 @@
+package org.kainos.ea.service;
+
```
Comment (mtslech) [inline product-backend/src/main/java/org/kainos/ea/service/BandValidator.java:2]: delete empty spaces

**Review by mtslech — APPROVED** [review]
L00kin g00d

**Pull request merged**